server/impl.ts

```diff
@@ -15,57 +15,36 @@ import {
 
 import { State, Logic, Player, DELTA, STEP, VELOCITY, COLORS } from './shared'
 
-type InternalState = GameState
-
-const delta = DELTA / 1000
-const step = STEP
-
-// NOTE: for some reason class properties don't seem to
-// be persisting across method calls; this is a workaround
-let idCounter = 0
-const state: {
+type InternalState = GameState & {
+	id: number
 	colorsBag: Color[]
 	accumulator: number
 	keepAlive: Map<string, number>
 	lastTimestamps: Map<string, number>
 	inputs: Map<string, Input>
-}[] = []
-
-function createState(id: number) {
-	state[id] = {
-		colorsBag: [...COLORS],
-		accumulator: 0,
-		keepAlive: new Map(),
-		lastTimestamps: new Map(),
-		inputs: new Map(),
-	}
-	return state[id]
 }
 
-function getState(id: number) {
-	return state[id] ?? createState(id)
-}
+const delta = DELTA / 1000
+const step = STEP
 
 export class Impl implements Methods<InternalState> {
 	initialize(ctx: Context, request: IInitializeRequest): InternalState {
-		const id = idCounter++
-		createState(id)
 		return {
-			id,
+			id: 0,
 			state: State.Empty,
 			time: 0,
 			startTime: 0,
 			players: [],
 			winner: '',
+			colorsBag: [...COLORS],
+			accumulator: 0,
+			keepAlive: new Map(),
+			lastTimestamps: new Map(),
+			inputs: new Map(),
```

Comment:
I see, this is much better :D I think that we don't need `id: 0` anymore

Reply:
done